src/main/java/co/cask/PluginJSON.java

```diff
@@ -177,6 +192,11 @@ public void execute()  throws MojoExecutionException {
       try {
         JSONTokener tokener = new JSONTokener(FileUtils.fileRead(file.getAbsoluteFile(), "UTF-8"));
         JSONObject object = new JSONObject(tokener);
+
+        JSONObject iconProperties = getIcon(iconFileNames, iconFiles, file);
+        if (iconProperties.length() != 0) {
```

Comment:
instead of this awkward check, rename getIcon to addIcon and pass in the object to add it to, where you can decide to add it if the file exists.